substrate-archive/src/tasks.rs

```diff
@@ -141,22 +151,31 @@ where
 		Self { api, backend, block, id }
 	}
 
-	fn block_into_storage(self) -> Result<BlockChanges<Block>, ArchiveError> {
-		let header = (&self.block).header();
+	fn prepare_block(
+		block: Block,
+		backend: &B,
+		id: &BlockId<Block>,
+	) -> Result<BlockPrep<Block, B::State, Block::Hash, NumberFor<Block>>, ArchiveError> {
+		let header = block.header();
 		let parent_hash = *header.parent_hash();
 		let hash = header.hash();
-		let num = *header.number();
+		let number = *header.number();
 
-		let state = self.backend.state_at(self.id)?;
+		let state = backend.state_at(*id)?;
 
 		// Wasm runtime calculates a different number of digest items
 		// than what we have in the block
 		// We don't do anything with consensus
 		// so digest isn't very important (we don't currently index digest items anyway)
 		// popping a digest item has no effect on storage changes afaik
-		let (mut header, ext) = self.block.deconstruct();
+		let (mut header, ext) = block.deconstruct();
 		header.digest_mut().pop();
-		let block = Block::new(header, ext);
+		Ok(BlockPrep { block: Block::new(header, ext), state, hash, parent_hash, number })
+	}
+
+	fn block_into_storage(self) -> Result<BlockChanges<Block>, ArchiveError> {
```

Comment:
```suggestion
	fn into_storage(self) -> Result<BlockChanges<Block>, ArchiveError> {
```
I don't think it really matters, but I think it would be nice to have the fn name start with `into_*`

Reply:
Actually, given that we have `execute_with_tracing` that does the same thing but with tracing (correct me if I'm wrong), it would nice if the names paired. So either `into_storage` & `into_storage_with_traces` or `execute` & `execute_with_tracing` etc